test/e2e/funcutils.go

```diff
@@ -0,0 +1,52 @@
+package e2e
+
+import (
+	"github.com/boson-project/func/test/cli"
+	"github.com/boson-project/func/test/e2e/utils"
+	"regexp"
+	"time"
+)
+
+// WaitFunctionToBecomeReady Waits a function to become ready. Timeout in 60 sec
+func WaitFunctionToBecomeReady(funcCli *cli.TestShellCli, functionName string) bool {
```

Comment:
This looks quite complicated. The wait loop can be done easily with some library. See e.g. here: https://github.com/openshift-knative/serverless-operator/blob/main/test/service.go#L255

Reply:
+1

Reply:
Agree. I'll consider it ! Thanks for the suggestion

Reply:
I'm just marking as resolved since there a new pr for e2e with a simplified logic as suggested